<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NativeAppPage</class>
 <widget class="QWidget" name="NativeAppPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>714</width>
    <height>518</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Firmware</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Project &amp;Target:</string>
        </property>
        <property name="buddy">
         <cstring>projectTargetRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout3">
        <item>
         <widget class="QRadioButton" name="projectTargetRadio">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ProjectTargetsComboBox" name="projectTarget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label2">
        <property name="text">
         <string>E&amp;xecutable:</string>
        </property>
        <property name="buddy">
         <cstring>executableRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout4">
        <item>
         <widget class="QRadioButton" name="executableRadio">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KUrlRequester" name="executablePath">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="placeholderText">
           <string>Enter the executable name or absolute path to an executable file</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QGroupBox" name="stackedWidgetPage1">
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="presetsLabel">
         <property name="text">
          <string>Presets:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="presetsComboPage1">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string>Arduino</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>OpenOCD</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="boardLabel">
         <property name="text">
          <string>Board:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="boardCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string>Nano</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="freqLabel">
         <property name="text">
          <string>MCU:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KComboBox" name="mcuCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="editable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="interfaceLabel">
         <property name="text">
          <string>Interface:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="KComboBox" name="interfaceCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Please, connect an interface</string>
         </property>
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="autoCompletion">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="interfaceBRLabel">
         <property name="text">
          <string>Interface baud rate:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="KComboBox" name="baudCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="autoCompletion">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="argumentsLabel">
         <property name="text">
          <string>Arguments:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="KComboBox" name="argumentsCombo">
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="autoCompletion">
          <bool>false</bool>
         </property>
         <item>
          <property name="text">
           <string>-C%avrdudeconf -v true -p%mcu -carduino -P%interface -b%baud -D -Uflash:w:%hex:i</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="workdirLabel">
         <property name="text">
          <string>Working Directory:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="KUrlRequester" name="workingDirectory">
         <property name="toolTip">
          <string>Select a working directory for the executable</string>
         </property>
         <property name="placeholderText">
          <string>Select a working directory for the executable</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="commandLabel">
         <property name="text">
          <string>Command:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="KComboBox" name="commandCombo">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;p&gt;Defines the command to execute the external terminal emulator. Use the following placeholders:&lt;/p&gt;
&lt;dl&gt;
&lt;dt&gt;&lt;code&gt;%exe&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the executable selected above.&lt;/dd&gt;
&lt;dt&gt;&lt;code&gt;%workdir&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the working directory selected above.&lt;/dd&gt;
&lt;/dl&gt;
&lt;p&gt;The arguments defined above will get appended to this command.&lt;/p&gt;</string>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="autoCompletion">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>%avrdude</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>/usr/bin/avrdude</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="8" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="stackedWidgetPage2">
      <layout class="QFormLayout" name="formLayout2">
       <item row="0" column="0">
        <widget class="QLabel" name="presetsLabel2">
         <property name="text">
          <string>Presets:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="presetsComboPage2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string>Arduino</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>OpenOCD</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="argumentsLabel2">
         <property name="text">
          <string>Arguments:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KComboBox" name="openocdArgumentsCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="autoCompletion">
          <bool>false</bool>
         </property>
         <item>
          <property name="text">
           <string>-f board/board.cfg</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>-f interface/interface.cfg -f target/target.cfg</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="workdirLabel2">
         <property name="text">
          <string>Working Directory:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="KUrlRequester" name="openocdWorkingDirectory">
         <property name="toolTip">
          <string>Select a working directory for the executable</string>
         </property>
         <property name="placeholderText">
          <string>Select a working directory for the executable</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="commandLabel2">
         <property name="text">
          <string>Command:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout2">
         <item>
          <widget class="KComboBox" name="openocdCommand">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;p&gt;Defines the command to execute the external terminal emulator. Use the following placeholders:&lt;/p&gt;
&lt;dl&gt;
&lt;dt&gt;&lt;code&gt;%exe&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the executable selected above.&lt;/dd&gt;
&lt;dt&gt;&lt;code&gt;%workdir&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the working directory selected above.&lt;/dd&gt;
&lt;/dl&gt;
&lt;p&gt;The arguments defined above will get appended to this command.&lt;/p&gt;</string>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="autoCompletion">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>%openocd</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>/usr/bin/openocd</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>ProjectTargetsComboBox</class>
   <extends>QComboBox</extends>
   <header>projecttargetscombobox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>executablePath</tabstop>
  <tabstop>workingDirectory</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>152</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>310</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
